docs/articles/concepts/kafka.md

```diff
@@ -0,0 +1,167 @@
+
+```python
+from bytewax.connectors.kafka import operators as kop, KafkaSinkMessage
+from bytewax import operators as op
+from bytewax.dataflow import Dataflow
+
+brokers = ["localhost:19092"]
+flow = Dataflow("example")
+kinp = kop.input("kafka-in", flow, brokers=brokers, topics=["in-topic"])
+processed = op.map("map", kinp.oks, lambda x: KafkaSinkMessage(x.key, x.value))
+kop.output("kafka-out", processed, brokers=brokers, topic="out-topic")
+```
+
+## Error handling
+
+The [`KafkaSource`](/apidocs/bytewax.connectors/kafka/index#bytewax.connectors.kafka.KafkaSource) input
+source will raise an exception whenever an error is encountered when consuming from Kafka.
+
+The Kafka input [operator](/apidocs/bytewax.connectors/kafka/operators#bytewax.connectors.kafka.operators.input)
+returns a dataclass containing two output streams. The `.oks` field is a stream of
+[`KafkaMessage`](/apidocs/bytewax.connectors/kafka/message#bytewax.connectors.kafka.message.KafkaSourceMessage)
+that were successfully processed. The `.errs` field is a stream of [`KafkaError`](/apidocs/bytewax.connectors/kafka/error#bytewax.connectors.kafka.message.KafkaError)
+messages where an error was encountered. Items that encountered an error have their `.err` field set with more
+details about the error.
+
+Note that if no processing is attached to the `.errs` stream of messages, they will be silently
+dropped, and processing will continue.
+
+Alternatively, error messages in the `.errs` stream can be published to a "dead letter queue" where they
+can be inspected and reprocessed later, while allowing the dataflow to continue processing data.
+
+## Batch sizes
+
+By default, Bytewax will consume a single message at a time from Kafka. The default
+setting is often sufficient for lower latency, but negatively affects throughput.
+
+If your dataflow would benefit from higher throughput, you can set the `batch_size` parameter
+to a higher value (eg: 1000). The `batch_size` parameter configures the maximum number of messages
+that will be fetched at a time from each worker.
+
+## Message Types
+
+Messages received from `KafkaSource` are of type [`KafkaSourceMessage`](
+/apidocs/bytewax.connectors/kafka/message#bytewax.connectors.kafka.message.KafkaSourceMessage).
+This dataclass includes basic Kafka fields like `.key` and `.value`, as well as
+extra information fields like `.headers`.
+
+Messages that are published to a `KafkaSink` must be of type
+[`KafkaSinkMessage`](/apidocs/bytewax.connectors/kafka/message#bytewax.connectors.kafka.message.KafkaSinkMessage).
+With the `.key` and `.value` fields set, or of type `KafkaSourceMessage`, if the message is unmodified.
+
+## Kafka and Recovery
+
+Typical deployments of Kafka utilize [consumer groups](https://developer.confluent.io/courses/architecture/consumer-group-protocol/)
+in order to manage partition assignment and the storing of Kafka offsets.
```

Comment:
Up until reading this, I was 99% sure that even when there's a single consumer it gets assigned to its own consumer group. Skimming through the docs I don't see any explicit confirmation for that, but I also don't see anything that hints otherwise. I think we should soften this particular statement, with something like

> Typically, in order to manage partition assignment and the storing of Kafka offsets, every consumer is assigned to a consumer group (for single consumer case it gets assigned to one automatically)

Also, I think we can link this https://developer.confluent.io/learn/apache-kafka-on-the-go/consumer-groups/ instead, or have both links (the one you provided is more technical, so we can have a separate mention about it like "If you are interested in learning about how Kafka uses consumer groups, check this link")